From 8f9bab4a910e14b62ba501acbd86016ff1d624c8 Mon Sep 17 00:00:00 2001
From: Eric Van Norman <ericvn@us.ibm.com>
Date: Thu, 24 Mar 2022 15:01:02 -0500
Subject: Update to do chmod on files after copy. Add read for non ELF files.
 (#38111)

* Update to do chmod on files after copy. Add read for non ELF files.

* Fix missing chmods and add recursion for directories
---
 tools/docker-copy.sh | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/tools/docker-copy.sh b/tools/docker-copy.sh
index 6ba1d15a53..1686915ece 100755
--- a/tools/docker-copy.sh
+++ b/tools/docker-copy.sh
@@ -32,16 +32,18 @@ function may_copy_into_arch_named_sub_dir() {
   #   arm64/
   #   amd64/
   if [[ ${FILE_INFO} == *"ELF 64-bit LSB"* ]]; then
-    chmod 755 "${FILE}"
     case ${FILE_INFO} in
       *x86-64*)
         mkdir -p "${DOCKER_WORKING_DIR}/amd64/" && cp -rp "${FILE}" "${DOCKER_WORKING_DIR}/amd64/"
+        chmod 755 "${DOCKER_WORKING_DIR}/amd64/$(basename "${FILE}")"
         ;;
       *aarch64*)
         mkdir -p "${DOCKER_WORKING_DIR}/arm64/" && cp -rp "${FILE}" "${DOCKER_WORKING_DIR}/arm64/"
+        chmod 755 "${DOCKER_WORKING_DIR}/arm64/$(basename "${FILE}")"
         ;;
       *)
         cp -rp "${FILE}" "${DOCKER_WORKING_DIR}"
+        chmod 755 "${DOCKER_WORKING_DIR}/$(basename "${FILE}")"
         ;;
     esac
 
@@ -61,8 +63,14 @@ function may_copy_into_arch_named_sub_dir() {
     fi
 
   else
-    chmod 644 "${FILE}"
     cp -rp "${FILE}" "${DOCKER_WORKING_DIR}"
+    file="${DOCKER_WORKING_DIR}/$(basename "${FILE}")"
+    if [[ -d ${file} ]]
+    then
+      chmod -R a+r "${file}"
+    else
+      chmod a+r "${file}"
+    fi
   fi
 }
 
-- 
2.35.3

